front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -170,13 +143,40 @@ async function handler(
         });
       }
 
-      const dataSource = await DataSource.create({
+      let dataSource = await DataSource.create({
         name: dataSourceName,
         description: dataSourceDescription,
         visibility: "private",
         config: JSON.stringify(dustDataSource.value.data_source.config),
         dustAPIProjectId: dustProject.value.project.project_id.toString(),
         workspaceId: owner.id,
+      });
+
+      const connectorsRes = await ConnectorsAPI.createConnector(
+        provider,
+        owner.sId,
+        systemAPIKeyRes.value.secret,
+        dataSourceName,
+        req.body.nangoConnectionId
+      );
+      if (connectorsRes.isErr()) {
+        logger.error(
```

Comment:
As per IRL: let's delete the datasource if it fails.

Reply:
Done